### DISCUSSION TITLE
xstate for Dart / Flutter?

### DISCUSSION THREAD

**Discussion by jinalskothari** [statelyai/xstate#3420] [Ideas] [answered]
Hello!

xstate is such a powerful paradigm, it makes certain things orders of magnitudes better. Flutter / Dart is a growing ecosystem, more people are getting into it. There is nothing comparable to xstate in Dart.

What do people think of an official port?

**Comment by davidkpiano** [comment, marked as answer]
That would be great!

I think the plan would be to start with porting XState FSM over, and then eventually the full XState API.

Also, please also post this as a feature request in https://statelyai.canny.io

**Reply by jinalskothari** [reply]
Great :)

Done
https://statelyai.canny.io/xstate/p/port-xstate-to-dart-flutter

**Comment by phamnhuvu-dev** [comment]
XState could be visualized as awesome, I really want it when I research how to modeling application logic, I could not do that with redux now
